encoding/protobuf.go

```diff
@@ -20,6 +21,53 @@ type protoSerializer struct {
 	serviceName string
 	methodName  string
 	method      *desc.MethodDescriptor
+	anyResolver jsonpb.AnyResolver
+}
+
+// a type to wrap a raw byte slice. Especially useful for any
+// types where we can't find the value in the registry.
+type bytesMsg struct {
+	V []byte
+}
+
+
+// A custom any resolver which will use the descriptor provider
+// and fallback to a simple byte array structure in the json.
+// This is the needed otherwise the default behaviour of protoreflect
+// is to fail with an error when it can't find a type.
+type anyResolver struct {
+	source protobuf.DescriptorProvider
+}
+
+func (*bytesMsg) ProtoMessage()             {}
+func (*bytesMsg) XXX_WellKnownType() string { return "BytesValue" }
+func (m *bytesMsg) Reset()                  { *m = bytesMsg{} }
+func (m *bytesMsg) String() string {
+	return fmt.Sprintf("%x", m.V) // not compatible w/ pb oct
+}
+func (m *bytesMsg) Unmarshal(b []byte) error {
+	m.V = append([]byte(nil), b...)
+	return nil
+}
+
+func (r anyResolver) Resolve(typeUrl string) (proto.Message, error) {
+	mname := typeUrl
+	if slash := strings.LastIndex(mname, "/"); slash >= 0 {
+		mname = mname[slash+1:]
+	}
```

Comment:
Looking at the testcases, maps the typeurls like
`schemas.test.proto/3ae3e282-1a1f-4921-91b4-12369cfc6036/Foo` to a message. This seems like a different pattern than the default according to https://developers.google.com/protocol-buffers/docs/proto3#any, 
> The default type URL for a given message type is type.googleapis.com/_packagename_._messagename_.

I think it would be good to add a comment here indicating what type of typeurl patterns this resolver intends to support.